istioctl/cmd/proxyconfig.go

```diff
@@ -232,14 +233,19 @@ func setupEnvoyServerStatsConfig(podName, podNamespace string, outputFormat stri
 		return "", fmt.Errorf("failed to create Kubernetes client: %v", err)
 	}
 	path := "stats"
+	port := 15000
 	if outputFormat == jsonOutput || outputFormat == yamlOutput {
 		// for yaml output we will convert the json to yaml when printed
 		path += "?format=json"
-	}
-	if outputFormat == prometheusOutput {
+	} else if outputFormat == prometheusOutput {
+		path += "/prometheus"
+	} else if outputFormat == prometheusMergedOutput {
 		path += "/prometheus"
+		port = 15020
```

Comment:
15020 is  configured via meshconfig StatusPort right?  Do we know if users modify this value frequently?  Does it need to be exposed as a flag passed into the `istioctl x envoy-stats` command?  Granted, 15000 is also configurable in meshconfig and I don't think it is exposed for other istioctl commands

Reply:
we should read this from pod annotations?

Reply:
15000 or 15020? Either way, since the other commands do not have a way to configure a port, the fix/feature is probably outside the scope of this PR and can be addressed in a separate PR or part of a possible design doc